openquake/calculators/scenario_damage.py

```diff
@@ -89,28 +88,43 @@ def scenario_damage(riskinputs, riskmodel, rlzs_assoc, monitor):
     :param monitor:
         :class:`openquake.baselib.performance.PerformanceMonitor` instance
     :returns:
-        a dictionary {('asset', asset): <mean stddev>,
-                      ('taxonomy', asset.taxonomy): <damage array>}
+        a dictionary {'asset': [(l, r, a, mean-stddev), ...],
+                      'taxonomy': damage array of shape T, L, R, E, D}
     """
     logging.info('Process %d, considering %d risk input(s) of weight %d',
                  os.getpid(), len(riskinputs),
                  sum(ri.weight for ri in riskinputs))
+    c_models = monitor.consequence_models
     L = len(riskmodel.loss_types)
     R = len(rlzs_assoc.realizations)
-    # D = len(riskmodel.damage_states)
+    D = len(riskmodel.damage_states)
+    E = monitor.oqparam.number_of_ground_motion_fields
+    T = len(monitor.taxonomies)
     taxo2idx = {taxo: i for i, taxo in enumerate(monitor.taxonomies)}
     lt2idx = {lt: i for i, lt in enumerate(riskmodel.loss_types)}
-    result = calc.build_dict((L, R), AccumDict)
+    result = dict(d_asset=[], d_taxon=numpy.zeros((T, L, R, E, D), F64),
+                  c_asset=[], c_taxon=numpy.zeros((T, L, R, E), F64))
     for out_by_rlz in riskmodel.gen_outputs(
             riskinputs, rlzs_assoc, monitor):
         for out in out_by_rlz:
-            lti = lt2idx[out.loss_type]
+            l = lt2idx[out.loss_type]
+            r = out.hid
+            c_model = c_models.get(out.loss_type)
             for asset, fraction in zip(out.assets, out.damages):
+                t = taxo2idx[asset.taxonomy]
                 damages = fraction * asset.number
-                result[lti, out.hid] += {
-                    ('asset', asset.idx): scientific.mean_std(damages)}
-                result[lti, out.hid] += {
-                    ('taxon', taxo2idx[asset.taxonomy]): damages}
+                if c_model:  # compute consequences
+                    means = [par[0] for par in c_model[asset.taxonomy].params]
+                    # NB: below we add a 0 in from for no damage state
+                    c_ratio = numpy.dot(fraction, [0] + means)
+                    consequences = c_ratio * asset.value(out.loss_type)
```

Comment:
It would be helpful to add a comment here that the consequences for occupants will need to be calculated separately, when we add that in the future.